invisible_cities/io/hist_io.py

```diff
@@ -27,3 +27,36 @@ def write_hist(histo : 'np.array: RWF, CWF, SiPM'):
         hist_table.append(histo.reshape(1, n_sensors, n_bins))
 
     return write_hist
+
+def hist_writer_var(file, *, compression='ZLIB4'):
+
+    def write_hist(group_name  : 'string with folder name to save histograms',
+                   table_name  : 'histogram name'                            ,
+                   entries     : 'np.array with bin content'                 ,
+                   bins        : 'list of np.array of bins'                  ,
+                   out_of_range: 'np.array lenght=2 with events out of range',
+                   errors      : 'np.array with bins uncertainties'          ,
+                   labels      : 'list with labels of the histogram'):
+
+        try:                       hist_group = getattr          (file.root, group_name)
+        except tb.NoSuchNodeError: hist_group = file.create_group(file.root, group_name)
+
+        if table_name in hist_group:
+            raise ValueError("Histogram {} already exists".format(table_name))
+
+        vlarray = file.create_vlarray(hist_group, table_name + '_bins', atom=tb.Float64Atom(shape = ()), filters=tbl.filters(compression))
+        for ibin in bins:
+            vlarray.append(ibin)
+        add_carray       (hist_group, table_name                   , entries     )
+        add_carray       (hist_group, table_name + '_outRange'     , out_of_range)
+        add_carray       (hist_group, table_name + '_errors'       , errors      )
+        file.create_array(hist_group, table_name + '_labels'       , labels      )
```

Comment:
If I recall correctly, because it is not a numpy array. Doing a numpy array of that so the add_carray function works would return a numpy array of dtype object I believe.